tests/gdtest/gdtest.c

```diff
@@ -21,7 +23,14 @@
 #endif
 
 #ifdef _WIN32
-# include "readdir.h"
+#include "readdir.h"
+#include <errno.h>
+#endif
+
+/* GDTEST_TOP_DIR is defined in other compile ways except msys
+ * test_config.h is created by windows/msys/run_test.sh*/
+#ifndef GDTEST_TOP_DIR
+#include <test_config.h>
 #endif
```

Comment:
`GDTEST_TOP_DIR`  is used in function `gdTestFilePathV` and is defined in `test_config.h`, which is generated by `windows/msys/run_tests.sh`. But this file is not included by gdtest.c.

Reply:
The other way is similar
```
#ifdef __MINGW32__
#incude <test_config.h>
#endif
```